File: order-service/src/main/java/com/intexsoft/webshop/orderservice/service/impl/StatusServiceImpl.java
package com.intexsoft.webshop.orderservice.service.impl;

import com.intexsoft.webshop.orderservice.dto.status.StatusDto;
import com.intexsoft.webshop.orderservice.mapper.StatusMapper;
import com.intexsoft.webshop.orderservice.model.Order;
import com.intexsoft.webshop.orderservice.model.Status;
import com.intexsoft.webshop.orderservice.model.enums.StatusName;
import com.intexsoft.webshop.orderservice.repository.OrderRepository;
import com.intexsoft.webshop.orderservice.repository.StatusRepository;
import com.intexsoft.webshop.orderservice.service.StatusService;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
@Slf4j
public class StatusServiceImpl implements StatusService {
    private final StatusRepository statusRepository;
    private final OrderRepository orderRepository;
    private final StatusMapper statusMapper;

    @Override
    public StatusDto addStatusToOrderWithId(Long orderId, StatusName statusName) {
        log.info("IN: try to add new status = {} to order with id = {}", statusName, orderId);
        Order foundOrder = orderRepository.findById(orderId).orElseThrow(
                () -> new RuntimeException("Order with id= " + orderId + " not found"));
        Status addedStatus = statusRepository.save(statusMapper.toStatus(foundOrder, statusName));
        log.info("IN: status = {} successfully added to order with id = {}", addedStatus.getStatusName(), orderId);
        return statusMapper.toStatusDto(addedStatus);
    }
}

File: order-service-kt/src/main/kotlin/com/intexsoft/webshop/orderservicekt/service/impl/StatusServiceImpl.kt
package com.intexsoft.webshop.orderservicekt.service.impl

import com.intexsoft.webshop.orderservicekt.dto.status.StatusDto
import com.intexsoft.webshop.orderservicekt.log
import com.intexsoft.webshop.orderservicekt.mapper.StatusMapper
import com.intexsoft.webshop.orderservicekt.model.Order
import com.intexsoft.webshop.orderservicekt.model.Status
import com.intexsoft.webshop.orderservicekt.model.enums.StatusName
import com.intexsoft.webshop.orderservicekt.repository.OrderRepository
import com.intexsoft.webshop.orderservicekt.repository.StatusRepository
import com.intexsoft.webshop.orderservicekt.service.StatusService
import org.springframework.data.repository.findByIdOrNull
import org.springframework.stereotype.Service

@Service
class StatusServiceImpl(
    private val statusRepository: StatusRepository,
    private val orderRepository: OrderRepository,
    private val statusMapper: StatusMapper
) : StatusService {

    override fun addStatusToOrderWithId(orderId: Long, statusName: StatusName): StatusDto {
        log.info("IN: try to add new status = $statusName to order with id = $orderId")
        val foundOrder: Order =
            orderRepository.findByIdOrNull(orderId) ?: throw RuntimeException("Order with id= $orderId not found")
        val addedStatus: Status = statusRepository.save(statusMapper.toStatus(foundOrder, statusName))
        log.info("IN: status = ${addedStatus.statusName} successfully added to order with id = $orderId")
        return statusMapper.toStatusDto(addedStatus)
    }
}
